docs: Move N4A FAQ content to docs (#537)
nginx/documentation: content/nginxaas-azure/overview/overview.md

NGINX as a Service for Azure is a service offering that is tightly integrated into Microsoft Azure public cloud and its ecosystem, making applications fast, efficient, and reliable with full lifecycle management of advanced NGINX traffic services.
NGINXaaS for Azure is available in the Azure Marketplace.

NGINXaaS for Azure is powered by [NGINX Plus](https://www.nginx.com/products/nginx/), which extends NGINX Open Source with advanced functionality and provides customers with a complete application delivery solution. Initial use cases covered by NGINXaaS include L4 TCP and L7 HTTP load balancing and reverse proxy which can be managed through various Azure management tools.
NGINXaaS allows you to provision distinct deployments as per your business or technical requirements.

> NGINXaaS handles the NGINX Plus license management automatically

Reviewer: ```suggestion
NGINXaaS handles the NGINX Plus license management automatically.
```

- Azure management tools (API, CLI, portal, terraform) work with NGINXaaS to create, update, and delete deployments
- Each NGINXaaS deployment has dedicated network and compute resources. There is no possibility of [noisy neighbor problems](https://learn.microsoft.com/en-us/azure/architecture/antipatterns/noisy-neighbor/noisy-neighbor) or data leakage between deployments

> - NGINXaaS can load balance even if upstream servers are located in different geography as long as no networking limitations are mentioned in the [Known Issues]({{< ref "/nginxaas-azure/known-issues.md" >}})

Reviewer: ```suggestion
- NGINXaaS can route traffic to upstreams even if the upstream servers are located in different geographies. See [Known Issues]({{< ref "/nginxaas-azure/known-issues.md" >}}) for any networking restrictions.
```

- NGINXaaS supports request tracing. See the [Application Performance Management with NGINX Variables](https://www.f5.com/company/blog/nginx/application-tracing-nginx-plus) blog to learn more about tracing.
- Supports HTTP to HTTPS, HTTPS to HTTP, and HTTP to HTTP redirects. NGINXaaS also provides the ability to create new rules for redirecting. See [How to Create NGINX Rewrite Rules | NGINX](https://www.nginx.com/blog/creating-nginx-rewrite-rules/) for more details.

> - NGINXaaS supports VNet, and VPN gateways if they do not have limitations. Known limitations can be found in the [Known Issues]({{< ref "/nginxaas-azure/known-issues.md" >}}).

Reviewer: ```suggestion
- NGINXaaS supports VNet, and VPN gateways if they do not have limitations. Known limitations can be found in the [Known Issues]({{< ref "/nginxaas-azure/known-issues.md" >}}).
```

Have tried out VPN gateways at all? if not, maybe redact this until we do? I remember we had issues with epxressroutes once so calling out non-test technologies may not be a good idea.

Author: This is in the FAQ in the production docs:
https://docs.nginx.com/nginxaas/azure/faq/#:~:text=What%20are%20the,Known%20Issues.

Author: The scope of this PR is to move the content from the FAQ to relevant docs. We can:

- Move it somewhere else more relevant.
- Leave it as is in this PR and work on improving it separately.
- Delete it.

Author: ```suggestion
- NGINXaaS is deployed inside of your Azure network and can connect to your upstream application running in your ecosystem. Known networking limitations can be found in the [Known Issues]({{< ref "/nginxaas-azure/known-issues.md" >}}).
```

Reviewer: 👍

{{< img src="nginxaas-azure/n4a-data-plane-architecture.svg" alt="The diagram illustrates the architecture of F5 NGINXaaS for Azure, showing end users accessing a public IP that routes through a network security group within a customer's Azure subscription. This leads to a delegated subnet in a virtual network, which connects to a zone-redundant load balancer within the NGINXaaS subscription. The load balancer distributes traffic across NGINX Plus instances in multiple availability zones, ensuring scalability and redundancy." >}}

> - NGINXaaS uses new Azure networking capabilities to keep end-user traffic private. Each NGINX Plus instance passes traffic to  downstream services using an elastic network card (NIC) that exists inside your subscription. These NICs are injected into a delegated virtual network. A network security group ensures that the deployment is secured and inbound connections are allowed to the ports the NGINX service listens to.

Reviewer 2: I would be in favor of removing all references to NICs or eNICs here. The eNIC is no longer visible in the customer's subscription and this section risks confusing them.
```suggestion
- NGINXaaS uses new Azure networking capabilities to keep end-user traffic private. A network security group ensures that the deployment is secured and inbound connections are allowed to the ports the NGINX service listens to.
```

- You can restrict access to NGINXaaS by defining restriction rules at the Network Security Group level or using NGINX's access control list. To learn more, see the [NGINX module ngx_http_access_module](http://nginx.org/en/docs/http/ngx_http_access_module.html) documentation.

> - NGINX Plus instances are automatically upgraded to receive security patches and the latest stable NGINX Plus version.

Reviewer: ```suggestion
- NGINXaaS deployment is automatically upgraded to receive security patches and the latest stable NGINX Plus version.
```
